src/CssParsing/StyleRule.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\CssParsing;
```

Comment:
I prefer to name classes based on what their objects represent, rather than how they might be used.

So I would go for just `Css`.  We could also move the `CssDocument` class into this namespace, renaming it to just `Document`.  (It is currently only in `Utilities` because the wasn't an obvious namespace for it at the time.)

Reply:
This then brings about the question: couldn't `CssInliner` be called just `Inliner` and exist in the `Css` namespace?

It doesn't feel like it belongs there, or that it should be renamed such.  It's an active class.  These other classes are passive, data only.  So perhaps the namespace should be `Data\Css` - though I don't like `Data` as a namespace name, it seems (for want of a better word) naff.

Reply:
> I prefer to name classes based on what their objects represent, rather than how they might be used.

Just to clarify, active classes can be named based on whay they (themselves) do, whereas passive (data-only) classes should be named based on what they represent (or store as data).

Reply:
Moved.

Concerning `CssInliner`: The class does not represent something that "is" CSS (like the other classes in the `Css` namespace). However, it does something with the HTML, and hence would belong into the `HtmlProcessor` namespace, I think.

Reply:
> However, it does something with the HTML, and hence would belong into the `HtmlProcessor` namespace, I think.

I think so too; the other classes there are like its siblings.  However, moving it would have a backwards compatibility issue.  Given #994 pending, I would avoid making such a change right now.  (I would also be keen to put in backwards-compatibility workarounds for any otherwise breaking change whenever possible - in the case of a renamed class, that would be a class with the old name that simply extends the newly renamed class - but wouldn't want such workarounds to pile up unnecessarily by renaming things on a whim without a clear plan.)